pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -209,6 +216,15 @@ func (plugin *kubenetNetworkPlugin) SetUpPod(namespace string, name string, id k
 
 	plugin.podCIDRs[id] = res.IP4.IP.String()
 
+	// Put the container bridge into promiscuous mode to force it to accept hairpin packets.
+	// TODO: Remove this once the kernel bug (#20096) is fixed.
+	if plugin.promisc {
+		// TODO: set with netlink once vishvananda/netlink gets that capability
+		if err := exec.Command("ip", "link", "set", BridgeName, "promisc", "on").Run(); err != nil {
```

Comment:
I'll add one; we check if the bridge exists before calling AddNetwork() and after, if it didn't previously exist we set promisc mode on it.  Otherwise we leave it alone.